### ISSUE TITLE
nonminimal_bool needs to make sure there is a space before the replaced expr

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#14988]
Using the following flags

`--force-warn clippy::nonminimal_bool`

this code:

```rust
fn main() {
	while(true && false) {}
}

```

caused the following diagnostics:

```
    Checking _main v0.1.0 (/tmp/icemaker_global_tempdir.AMunhyvPOeAJ/icemaker_clippyfix_tempdir.xO7o9pdlwjYp/_main)
warning: this boolean expression can be simplified
 --> src/main.rs:2:7
  |
2 |     while(true && false) {}
  |          ^^^^^^^^^^^^^^^ help: try: `false`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
  = note: requested on the command line with `--force-warn clippy::nonminimal-bool`

warning: `_main` (bin "_main") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.14s

```

However after applying these diagnostics, the resulting code:

```rust
fn main() {
	whilefalse {}
}

```
no longer compiled:

```
    Checking _main v0.1.0 (/tmp/icemaker_global_tempdir.AMunhyvPOeAJ/icemaker_clippyfix_tempdir.xO7o9pdlwjYp/_main)
error[E0422]: cannot find struct, variant or union type `whilefalse` in this scope
 --> src/main.rs:2:2
  |
2 |     whilefalse {}
  |     ^^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0422`.
error: could not compile `_main` (bin "_main" test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_main` (bin "_main") due to 1 previous error

```

Version:
```
rustc 1.89.0-nightly (cf423712b 2025-06-05)
binary: rustc
commit-hash: cf423712b9e95e9f6ec84b1ecb3d125e55ac8d56
commit-date: 2025-06-05
host: x86_64-unknown-linux-gnu
release: 1.89.0-nightly
LLVM version: 20.1.5

```
